### DISCUSSION TITLE
How to configure nativewind for use with storybook in React Native Expo

### DISCUSSION THREAD

**Discussion by jadrdev** [nativewind/nativewind#272] [General]
**Describe the bug**
ERROR in ./node_modules/react-native/index.js 14:7
Module parse failed: Unexpected token (14:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|
| // Components
> import typeof AccessibilityInfo from './Libraries/Components/AccessibilityInfo/AccessibilityInfo';
| import typeof ActivityIndicator from './Libraries/Components/ActivityIndicator/ActivityIndicator';
| import typeof Button from './Libraries/Components/Button';
 @ ./node_modules/nativewind/dist/expo-snack.js 5:23-46
 @ ./node_modules/nativewind/dist/index.js
 @ ./stories sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.(js|jsx|ts|tsx))$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/docs/config-generated-config-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-links/preview.js-generated-config-entry.js ./node_modules/@storybook/addon-docs/preview.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/preview.js-generated-config-entry.js ./node_modules/@storybook/addon-backgrounds/preview.js-generated-config-entry.js ./node_modules/@storybook/addon-measure/preview.js-generated-config-entry.js ./node_modules/@storybook/addon-outline/preview.js-generated-config-entry.js ./node_modules/@storybook/addon-interactions/preview.js-generated-config-entry.js ./.storybook/preview.js-generated-config-entry.js ./generated-stories-entry.js

StoryBook Configure
module.exports = {
  "stories": [
    "../stories/**/*.stories.mdx",
    "../stories/**/*.stories.@(js|jsx|ts|tsx)"
  ],
  "addons": [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-interactions"
  ],
  "framework": "@storybook/react"
}

babel configuration

module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: ["nativewind/babel"],
  };
};

**Comment by marklawlor** [comment]
I don't use storybook, but that is a common error you recieve when trying to use the untranspiled version of React Native. I would suggest looking at the Storybook documentation and consider using their official React Native / React Native Web plugins

https://github.com/storybookjs/react-native
https://storybook.js.org/addons/@storybook/addon-react-native-web

I'm going to move this to a discussion where other people could assist you, but if you any issues you should post them on the storybook repo.

**Comment by jadrdev** [comment]
In the end I solved it by installing react native web Storybook dependencies and adding the following line to my main.js

"@storybook/addon-react-native-web"

Thank you

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
